framework/src/relationshipmanagers/GhostBoundary.C

```diff
@@ -0,0 +1,125 @@
+
+  // We want to wait until our mortar mesh has been built before trying to delete remote elements.
+  // And our mortar mesh cannot be built until the entire mesh has been generated. By setting this
+  // parameter to false we will make sure that any prepare_for_use calls during the mesh generation
+  // phase will not delete remote elements *and* we will set a flag on the moose mesh saying that we
+  // need to delete remote elements after the addition of late geometric ghosting functors
+  // (including this ghosting functor)
+
+  return params;
+}
+
+GhostBoundary::GhostBoundary(const InputParameters & params)
+  : RelationshipManager(params),
+    _boundary_name(getParam<std::vector<BoundaryName>>("boundary")),
+    _ghost_point_neighbors(getParam<bool>("ghost_point_neighbors"))
+{
+}
+
+GhostBoundary::GhostBoundary(const GhostBoundary & other)
+  : RelationshipManager(other),
+    _boundary_name(other._boundary_name),
+    _ghost_point_neighbors(other._ghost_point_neighbors)
+{
+}
+
+void
+GhostBoundary::internalInitWithMesh(const MeshBase &)
+{
+}
+
+std::string
+GhostBoundary::getInfo() const
+{
+  std::ostringstream oss;
+  oss << "GhostBoundary";
+  return oss.str();
+}
+
+void
+GhostBoundary::operator()(const MeshBase::const_element_iterator & /*range_begin*/,
+                          const MeshBase::const_element_iterator & /*range_end*/,
+                          const processor_id_type p,
+                          map_type & coupled_elements)
+{
+  // We ask the user to pass boundary names instead of ids to our constraint object.  However, We
+  // are unable to get the boundary ids from boundary names until we've attached the MeshBase object
+  // to the MooseMesh
+  const bool generating_mesh = !_moose_mesh->getMeshPtr();
+  const auto boundary_ids = generating_mesh ? std::vector<BoundaryID>{Moose::INVALID_BOUNDARY_ID}
+                                            : _moose_mesh->getBoundaryIDs(_boundary_name);
+
+  for (const Elem * const elem : _mesh->active_element_ptr_range())
+  {
+    if (generating_mesh)
+    { // We are still generating the mesh, so it's possible we don't even have the right boundary
+      // ids created yet! So we actually ghost all boundary elements and all lower dimensional
+      // elements who have parents on a boundary
+      if (elem->on_boundary())
+        coupled_elements.insert(std::make_pair(elem, _null_mat));
+    }
+    else
+    {
+      // We've finished generating our mesh so we can be selective and only ghost elements lying on
+      // our boundary
+      const BoundaryInfo & binfo = _mesh->get_boundary_info();
+      for (auto side : elem->side_index_range())
+        for (auto boundary_id : boundary_ids)
+          if ((elem->processor_id() != p) && (binfo.has_boundary_id(elem, side, boundary_id)))
+            goto countBreak;
+    countBreak:
+      coupled_elements.insert(std::make_pair(elem, _null_mat));
```

Comment:
```suggestion
          {
            coupled_elements.insert(std::make_pair(elem, _null_mat));
            goto countBreak;
          }
      countBreak:
```

Reply:
As this is currently written, this will ghost every element

Reply:
I don't see the issue, unless the alignment was off. As it was initially once the `if` identifies an element with a valid boundary, exits both loops. the only problem might be if the element might be null, which I only checked via overall testing.  I changed it now but it looks funkier.  Btw, I added one more test in framework.

Reply:
Before implementing my suggestion, you would do a double for loop over sides and boundary IDs, and if you met the `if` condition you would jump ahead to `countBreak:`. But imagine that you don't satisfy the condition, you just do your double loop to the finish. Well you still get to `countBreak:` regardless of whether you exited the double for loops or not. And then once you're there you would have inserted into your `coupled_elements` container irrespective of whatever happened in the double for loop, e.g. irrespective of whether `binfo.has_boundary_id(elem, side, boundary_id)`. So it's critical that the `coupled_elements` insertion happen within the double for loop and not outside of its scope

Reply:
> Btw, I added one more test in framework

That's great. It looks like it's not using `GhostBoundary` though. You need some object that will trigger its addition. Once it's there then you'll see the framework coverage rate increase

Reply:
Yeah I was aiming for that, but it seems pretty disconnected from the splitting business, and the PR it addresses. Not sure what kind of problem to design exactly, cause splitting was such a good fit.